OSX xcodebuild target in GitHub Action

csomor opened the pull request:
add our xcode projects to GI as well

csomor committed:
first attempt for xcodebuild target

csomor committed:
arch was missing for shell

csomor committed:
switching to self hosted runner and crossing archs and types

csomor committed:
wrong level

csomor committed:
trying to better debug self-hosted runner

csomor committed:
removing ARCH additions

csomor committed:
going back to GitHub machines, to find out whether things work properly there …

csomor committed:
adding iOS builds

csomor committed:
no code signing required

csomor committed:
adding xcpretty back as installed now also on self-hosted runner, adding explicit configuration

csomor committed:
mixed up project path

csomor committed:
moving partly to self-hosted again, giving in on iphone app w/o certificate, building lib only

csomor committed:
no deprecation warnings, adding explicit path to pcre config

the self-hosted runner had problems compiling pcre, perhaps it was picking up a config.h from somewhere else …

csomor committed:
the reason for self-hosted problems: the are stateful

trying to workaround self-hosted stateful runners … build dir in a separate and cleaned dir, this should allow coexistence with xcodebuild, without always having to clear the checkout dir ..

csomor committed:
workaround error for not yet existing working dir

csomor committed:
removing cd remnants

csomor committed:
reach for makefile.unx in repo not in builddir

csomor commented:
@vadz I'd only transfer the `ci_mac_xcode.yml` of course

vadz commented on .github/workflows/ci_mac_xcode.yml:
```diff
@@ -0,0 +1,118 @@
+
+    - name: Building
+      run: |
+        set -o pipefail && xcodebuild clean build ONLY_ACTIVE_ARCH=NO \
```
`set -o pipefail` shouldn't be necessary, it's already set for the default shell value.

csomor replied:
I didn't see that, saw `pipefail` only in the mac-tests part, in the defaults I only see

`shell: /usr/bin/arch -arch ${{ matrix.arch }} /bin/bash -l {0}`

so is `/bin/bash -l` already using this ?

vadz replied:
I didn't notice we overrode the default shell in `ci_mac.yml`, it would be better to use the default options there too, to be consistent with the [documented defaults](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell), i.e. `bash --noprofile --norc -eo pipefail {0}`.

csomor replied:
thanks, Committed

vadz commented on .github/workflows/ci_mac_xcode.yml:
```diff
@@ -0,0 +1,118 @@
+      with:
+        submodules: recursive
+
+    - name: Building
```
Very minor, but for consistency with all the other steps we should do
```suggestion
    - name: Build
```

vadz commented on .github/workflows/ci_mac_xcode.yml:
```diff
@@ -0,0 +1,118 @@
+          destination: 'generic/platform=iOS'
+          arch: x86_64
+          configuration: 'Debug'
+          project: build/osx/wxiphone.xcodeproj
```
Why don't we use the minimal sample project for this one?

csomor replied:
I cannot build an iOS app without a certificate and provisioning profile. And I'm hesitant to use mine via github secrets, because self-hosted runners apparently could pose a security risk. I thought about adding a certificate to the machine itself, but then again this could be misused as well, so while I don't yet what the best solutions could be, I scaled back to just building the lib for the time being.

vadz replied:
I see, thanks for the explanation.

vadz reviewed (commented):
Thanks, this will definitely be useful to have!

Should I merge (just the Xcode-related part of) this or will you do it?

csomor committed:
Update .github/workflows/ci_mac_xcode.yml

Co-authored-by: VZ <vz-github@zeitlins.org>

csomor committed:
using proper bash defaults, so explicit pipefail setting not needed anymore

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell

csomor commented:
> Should I merge (just the Xcode-related part of) this or will you do it?

@vadz Thanks, honestly I'd be glad if you could do, I don't know about how to properly squash/merge/cherry-pick properly with git, I would have copied just the file into master ...

but this now uncovered failing tests in iOS, `TEST_INVALID_COMBOBOX_ISEMPTY` cannot fail to compile because there is no combobox at all in iOS, but I don't know how to best make this test conditional

vadz commented:
I'm not sure about the iOS tests, do you mean that `wxUSE_COMBOBOX == 0` there? If so, we probably need to handle this in `tests/controls/comboboxtest.cpp` itself, as this is not really iOS-specific and could happen in the other ports too if this option is set to 0.

csomor commented:
> I'm not sure about the iOS tests, do you mean that `wxUSE_COMBOBOX == 0` there? 

yes, I'm turning it off in `iphone/chkconf.h`

> If so, we probably need to handle this in `tests/controls/comboboxtest.cpp` itself, as this is not really iOS-specific and could happen in the other ports too if this option is set to 0.

ok, should I just add a compiler error there, or a #error ?

csomor commented:
like this
```
...
#else

#ifdef TEST_INVALID_COMBOBOX_ISEMPTY
#error provoke failing here
#endif

#endif //wxUSE_COMBOBOX
```

vadz commented:
Yes, this should be fine, thanks.

vadz commented:
Oops, sorry, I've used a wrong PR number in the commit message, please disregard 13339425beb2e173ab575a871d95b9feaa53a9ba mentioned above.

But I still applied the changes here, i.e. I think this one can be closed now -- please let me know if I forgot/omitted anything.
